ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -317,11 +364,22 @@ boolean isCaughtUp() {
   void shutdown() throws InterruptedException, StoreException, IOException {
     if (running.get()) {
       running.set(false);
-      synchronized (lock) {
-        lock.notify();
+      boolean isPaused = isPaused();
+      if (isPaused) {
+        reentrantLock.lock();
+        paused.set(false);
```

Comment:
this is coz, when the hard delete wakes up after being signalled, we are within a while loop which checks for pause value. So it will go into indefinite await until some other thread signals it. So, had to set it to false here. Once we are done with the clean up, setting it back to true before persisting.